app/src/org/commcare/engine/resource/AndroidResourceManager.java

```diff
@@ -304,5 +305,6 @@ private long exponentionalRetryDelay(int numberOfRestarts) {
     public void clearUpgrade() {
         super.clearUpgrade();
         HiddenPreferences.setReleasedOnTimeForOngoingAppDownload((AndroidCommCarePlatform)platform, 0);
+        HiddenPreferences.setPreUpdateSyncNeeded(PrefValues.NO);
```

Comment:
Hm, only thing about this PR that I'm not sure about. This will unset the flag even if the current profile installed (not staged) had it set. 

That is likely totally fine (and likely difficult to work around), but wanted to quickly confirm with you that you don't see any problems with that behavior?

Reply:
Hey yeah, thanks for flagging. I do think that's fine since this flag is only relevant when there is an update staged.